DOCS-248 Add troubleshooting for creating mappings in SQL (#500)
hazelcast/hz-docs: docs/modules/sql/pages/troubleshooting.adoc

=== Could not find ReadFileFnProvider

Hazelcast cannot read the file format that you're trying to map to. This error could be thrown because you are using the slim distribution of Hazelcast Platform, which doesn't come with the ability to read the file format by default.

Use the xref:getting-started:editions.adoc[full distribution] of Hazelcast Platform or find the missing module and add it to your cluster members' classpaths.

=== No suitable FileSourceFactory found

Hazelcast cannot read from the specified file system.

To read from some file systems, you must add additional modules to your cluster members' classpaths. See xref:sql:mapping-to-a-file-system.adoc#supported-file-systems[Supported file systems].

=== java.net.UnknownHostException: path

Your mapping contains an invalid URL.

Make sure that all your URLs are resolvable.

== JSON Errors

This section provides troubleshooting for using JSON in SQL.

=== JSON_QUERY evaluated to multiple values

The JsonPath is matching multiple values, which results in invalid JSON.

> If you expect multiple values to be returned from the JsonPath expression, you should add a wrapper, such as the xref:sql:functions-and-operators.adoc#json_query[`WITH CONDITIONAL ARRAY WRAPPER`] clause, to wrap the matches in an array.

Reviewer: ```suggestion
If it's possible that the JsonPath expression matches multiple values in edge cases, you should add a wrapper, such as the xref:sql:functions-and-operators.adoc#json_query[`WITH CONDITIONAL ARRAY WRAPPER`] clause, to wrap the matches in an array.
```
